lint/lint.go

```diff
@@ -76,27 +57,30 @@ func (l *Linter) LintProtosWithRegistry(files []*descriptorpb.FileDescriptorProt
 // It uses the proto file path to determine which rules will
 // be applied to the request, according to the list of Linter
 // configs.
-func (l *Linter) run(req Request) (Response, error) {
+func (l *Linter) lintFileDescriptor(fd *desc.FileDescriptor) (Response, error) {
 	resp := Response{
-		FilePath: req.ProtoFile().Path(),
+		FilePath: fd.GetName(),
+		Problems: []Problem{},
```

Comment:
I don't think we need to initialize problem array here -- better to remove this line.

Reply:
In general, I want to make sure to send down `[]Problem{}` and not `nil`, because it makes testing easier (both `reflect.DeepEqual` and `cmp.Diff` consider empty slice to be unequal to nil).